### ISSUE TITLE
CI breaks against GHC 9.2.1 due to lattices-1.2.1.1

### ISSUE THREAD

**Issue by ocramz** [msakai/data-interval#34]
Error :

```
Algebra/PartialOrd.hs:47:43: error:
    • Data.Map.fold is gone. Use foldr.
cabal-3.6.2.0: Failed to build lattices-1.2.1.1. See the build log above for details.
```

I suggest importing a newer version of 'lattices', which doesn't use `Data.Map.fold`

**Comment by Bodigrim** [comment]
I've seen the error in CI logs, but I cannot reproduce it locally. Seems like an intermittent Cabal solver bug: [`lattices-1.2.1.1`](https://hackage.haskell.org/package/lattices-1.2.1.1) demands `containers >=0.3 && <0.6`, while GHC 9.2 ships with `containers-0.6.5.1`. And yet [build plan](https://github.com/msakai/data-interval/runs/4085308309?check_suite_focus=true#step:15:28) says:

```
Build profile: -w ghc-9.2.1 -O1
In order, the following would be built (use -v for more details):
 - call-stack-0.4.0 (lib) (requires download & build)
 - clock-0.8.2 (lib) (requires download & build)
 - colour-2.3.6 (lib) (requires download & build)
 - happy-1.20.0 (exe:happy) (requires download & build)
 - hashable-1.4.0.0 (lib) (requires download & build)
 - lattices-1.2.1.1 (lib:lattices) (requires download & build)
```

**Comment by Bodigrim** [comment]
CC @gbaz @fgaz, any idea what's happened with Cabal solver here?

**Comment by gbaz** [comment]
No issue with the solver. The CI "helpfully" generates a cabal.project with allow-newer on containers. Don't ask me why!

https://github.com/msakai/data-interval/runs/4085308309?check_suite_focus=true#step:14:36

**Comment by Bodigrim** [comment]
Ah, right, that's my bad: `haskell-ci-0.13.20211029` treats GHC 9.2.1 as unreleased and helpfully enables head.hackage and `allow-newer` for all packages. I should have used `haskell-ci-0.13.20211030`. Thanks @gbaz!

**Comment by Bodigrim** [comment]
Fixed by 6b62faf1dc4de5a8cad241a10eedf90fa4c2b825. Thanks @ocramz!
